src/app/data-client.ts

```diff
@@ -118,40 +118,43 @@ export class DataClient {
     if (limit) {
       dataReq.setLimit(limit);
     }
-    dataReq.setSortOrder(sortOrder ?? Order.ORDER_UNSPECIFIED);
-    dataReq.setLast(last);
+    if (sortOrder) {
+      dataReq.setSortOrder(sortOrder);
+    }
+    if (last) {
+      dataReq.setLast(last);
+    }
 
     const req = new dataPb.TabularDataByFilterRequest();
     req.setDataRequest(dataReq);
     req.setCountOnly(countOnly);
     req.setIncludeInternalData(includeInternalData);
 
-    // eslint-disable-next-line no-await-in-loop
     const response = await promisify<
       dataPb.TabularDataByFilterRequest,
       dataPb.TabularDataByFilterResponse
     >(service.tabularDataByFilter.bind(service), req);
     const mdListLength = response.getMetadataList().length;
 
-    const dataArray: TabularData[] = [];
-    dataArray.push(
-      ...response.getDataList().map((data) => {
-        const mdIndex = data.getMetadataIndex();
+    const data: TabularData[] = [];
+    data.push(
+      ...response.getDataList().map((tabData) => {
+        const mdIndex = tabData.getMetadataIndex();
         const metadata =
           mdListLength !== 0 && mdIndex >= mdListLength
             ? new dataPb.CaptureMetadata().toObject()
             : response.getMetadataList()[mdIndex]?.toObject();
         return {
-          data: data.getData()?.toJavaScript(),
+          data: tabData.getData()?.toJavaScript(),
           metadata,
-          timeRequested: data.getTimeRequested()?.toDate(),
-          timeReceived: data.getTimeReceived()?.toDate(),
+          timeRequested: tabData.getTimeRequested()?.toDate(),
+          timeReceived: tabData.getTimeReceived()?.toDate(),
         };
       })
     );
 
     return {
-      array: dataArray,
+      array: data,
```

Comment:
OHH MY BAD I meant this should be called `data` rather than `array`! Don't really care about the variable name, just the returned name. Same as below

Reply:
THAT MAKES A LOT OF SENSE. Hahaha!